base/src/com/google/idea/blaze/base/model/primitives/WorkspacePath.java

```diff
@@ -57,7 +57,7 @@ public WorkspacePath(String relativePath) {
   }
 
   public WorkspacePath(WorkspacePath parentPath, String childPath) {
-    this(parentPath.relativePath() + BLAZE_COMPONENT_SEPARATOR + childPath);
+    this((parentPath.isWorkspaceRoot() ? "" : (parentPath.relativePath() + BLAZE_COMPONENT_SEPARATOR)) + childPath);
```

Comment:
I think this might be better for readability:

```
this(parentPath.isWorkspaceRoot() ? childPath : parentPath.relativePath() + BLAZE_COMPONENT_SEPARATOR + childPath);
```

Reply:
Agreed. Updated and pushed.